File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque
def solution(n, computers):
    answer = 0
    visited=[0]*(n)
    def bfs(start):
        q=deque()
        q.append(start)
        visited[start]=1
        while q:
            x=q.popleft()
            for i in range(n):
                if computers[x][i]==1 and visited[i]==0:
                    q.append(i)
                    visited[i]=1
    for i in range(n):
        if not visited[i]:
            bfs(i)
            answer+=1
        
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    static boolean[] visited;
    public int solution(int n, int[][] computers) {
        visited = new boolean[n];
        int answer = 0;
        for(int i = 0 ; i <n;i++){
            if(!visited[i]){
                dfs(i,n,computers);
                answer++;
            }
        }
        return answer;
    }
    void dfs(int current, int n, int[][] computers){
        visited[current]=true;
        for(int i = 0 ; i < n;i++){
            if(i!=current&&computers[current][i]==1&&!visited[i]){
                dfs(i,n,computers);
            }
        }
    }
}